fuel-vm/src/interpreter/executors/main.rs

```diff
@@ -474,7 +476,10 @@ where
         Ok(())
     }
 
-    pub(crate) fn run(&mut self) -> Result<ProgramState, InterpreterError<S::DataError>> {
+    pub(crate) fn run(
+        &mut self,
+        gas_price: Word,
```

Comment:
It seems can be part of the `InterpreterParams` along with another data=)

Reply:
I guess this is a question I had about this api in general. Will each instance of the interpreter only use a single price? Do we never share an interpreter between blocks? That's why I went this route.

Reply:
We always create a new instance of the `Interpreter` for each transaction. We don't have plans to share interpreter, since it is faster to allocate a new memory for `Interpreter` instead of clearing manually the old one, based on our benchmarks=)

Reply:
Okay. I'm convinced that putting it on the `Interpreter` is a good idea, but I'm not going to put it on `InterpreterParams` because we have conversions from `ConsensusParams` that would need an additional parameter, if we just put it on `Interpreter` then we can keep those conversions.

I might rename `InterpreterParams` to `InterpreterConsensusParams` because that's a bit more accurate.

Reply:
Hmm. I'm having a similar problem with `Interpreter`. We have some `From` conversions for other types that don't have `gas_price` so I'm going have to replace a bunch of `from` methods with new functions. 

It might end up being a better interface long term to have it held on the `Interpreter`, but it's not obvious to me right now if that's the right design. 

I don't hate having the extra param on the execution methods for now.

Reply:
What about something like?

```
impl impl From<(ConsensusParameters, GasPrice)> for InterpreterParams {
```

Reply:
I thought about going with the tuple approach, but I don't feel like it is very idiomatic.

I'll give the constructor another shot.